### PR TITLE
chore: bumpenvs a8b9c02, remove some selective tensorflow imports [MLG-358]

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

This bumpenvs reflects the upgrade of our `environments` images to TensorFlow 2.11.0.
The example `gpt_titanic_estimator` image has been held back; it will be updated when MLG-442 is done.
Some references in the docs to specific versions of TensorFlow, ROCm, and PyTorch have been updated (not all because of this particular bumpenvs).
Selective imports of Keras optimizers - based on the installed version of TensorFlow - have been removed from many examples and test fixtures: these are cases where it is reasonable to assume that the latest version of TensorFlow we support, 2.11, is installed. This reverts some changes made in #6527.

## Test Plan

All automated tests should pass, including and especially those in `test-e2e-longrunning`.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/training/apis-howto/overview.rst [hunk 1]
```diff
@@ -68,15 +68,16 @@ TensorFlow 1 vs 2
...
 Determined supports both TensorFlow 1 and 2. The version of TensorFlow that is used for a particular
 experiment is controlled by the container image that has been configured for that experiment.
-Determined provides prebuilt Docker images that include TensorFlow 2.8, 1.15, and 2.7, respectively:
+Determined provides prebuilt Docker images that include TensorFlow 2.11, 1.15, and 2.8,
+respectively:
 
--  ``determinedai/environments:cuda-11.3-pytorch-1.12-tf-2.8-gpu-0.21.2`` (default)
+-  ``determinedai/environments:cuda-11.3-pytorch-1.12-tf-2.11-gpu-0.21.2`` (default)
 -  ``determinedai/environments:cuda-10.2-pytorch-1.7-tf-1.15-gpu-0.21.2``
 -  ``determinedai/environments:cuda-11.2-tf-2.7-gpu-0.21.2``
```
Comment (mpkouznetsov) [inline docs/training/apis-howto/overview.rst:76]: is this 2.7 or 2.8?

File: docs/training/apis-howto/overview.rst [hunk 2]
```diff
@@ -92,7 +93,7 @@ images.
 .. _rocm-support:
 
 Determined has experimental support for ROCm. Determined provides a prebuilt Docker image that
-includes ROCm 4.2, PyTorch 1.9 and Tensorflow 2.5:
+includes ROCm 5.0, PyTorch 1.10 and Tensorflow 2.7:
```
Comment (mpkouznetsov) [inline docs/training/apis-howto/overview.rst:96]: Good catch.

**Review by mpkouznetsov — APPROVED** [review]

**Comment by drh-determined-ai** [comment]
As of [7e04f55](https://github.com/determined-ai/determined/pull/6595/commits/7e04f5560ee4a5220540f1ee3305bbfbbe4dc63b), all long-running e2e tests pass, _except_ for `tests/cluster/test_master_restart.py::test_master_restart_with_queued`, which is a bug in `main`, so we will ignore it here.
There will be some minor commits after this comment that should not affect `test-e2e` or `test-e2e-longrunning`, so the latter will not be run again.
